lnwire/partial_sig.go

```diff
@@ -0,0 +1,197 @@
+// partial signature. This is used for the co-op close flow, as both sides have
+// already exchanged nonces, so they can send just the partial signature.
+type PartialSig struct {
+	// Sig is the 32-byte musig2 partial signature.
+	Sig btcec.ModNScalar
+}
+
+// NewPartialSig creates a new partial sig.
+func NewPartialSig(sig btcec.ModNScalar) PartialSig {
+	return PartialSig{
+		Sig: sig,
+	}
+}
+
+// Record returns the tlv record for the partial sig.
+func (p *PartialSig) Record() tlv.Record {
+	return tlv.MakeStaticRecord(
+		PartialSigRecordType, p, PartialSigLen,
+		partialSigTypeEncoder, partialSigTypeDecoder,
+	)
+}
+
+// partialSigTypeEncoder encodes a 32-byte musig2 partial signature as a TLV
+// value.
+func partialSigTypeEncoder(w io.Writer, val interface{}, buf *[8]byte) error {
+	if v, ok := val.(*PartialSig); ok {
+		sigBytes := v.Sig.Bytes()
+		if _, err := w.Write(sigBytes[:]); err != nil {
+			return err
+		}
+
+		return nil
+	}
+
+	return tlv.NewTypeForEncodingErr(val, "lnwire.PartialSig")
+}
+
+// Encode writes the encoded version of this message to the passed io.Writer.
+func (p *PartialSig) Encode(w io.Writer) error {
+	return partialSigTypeEncoder(w, p, nil)
+}
+
+// partialSigWithNonceTypeDecoder decodes a 98-byte musig2 extended partial
+// signature.
+func partialSigTypeDecoder(r io.Reader, val interface{}, buf *[8]byte,
+	l uint64) error {
+
+	if v, ok := val.(*PartialSig); ok && l == PartialSigLen {
+		var sBytes [32]byte
+		if _, err := io.ReadFull(r, sBytes[:]); err != nil {
+			return err
+		}
+
+		var s btcec.ModNScalar
+		s.SetBytes(&sBytes)
+
+		*v = PartialSig{
+			Sig: s,
+		}
+		return nil
+	}
+
+	return tlv.NewTypeForDecodingErr(val, "lnwire.PartialSig", l,
+		PartialSigLen)
+}
+
+// Decode reads the encoded version of this message from the passed io.Reader.
+func (p *PartialSig) Decode(r io.Reader) error {
+	return partialSigTypeDecoder(r, p, nil, PartialSigLen)
+}
+
+const (
+	// PartialSigWithNonceLength is the length of a serialized
+	// PartialSigWithNonce. The sig is encoded as the 32 byte S value
+	// followed by the 66 nonce value.
+	PartialSigWithNonceLen = 98
+
+	// PartialSigWithNonceRecordType...
+	PartialSigWithNonceRecordType = 2
+)
+
+// PartialSigWithNonce is a partial signature with the nonce that was used to
+// generate the signature. This is used for funding as well as the commitment
+// transaction update dance. By sending the nonce only with the signature, we
+// enable the sender to generate their nonce just before they create their
+// signature. Signers can use this trait to mix in additional contextual data
+// such as the commitment txn itself into their nonce generation function.
+//
+// The final signature is 98 bytes: 32 bytes for the S value, and 66 bytes for
+// the public nonce (two compressed points).
+type PartialSigWithNonce struct {
+	PartialSig
+
+	// Nonce is the 66-byte musig2 nonce.
+	Nonce Musig2Nonce
```

Comment:
The `Musig2Nonce` type is only introduced in the next commit. So just gotta swap these two commits for this to compile